src/components/node-list/node-list-row.js

```diff
@@ -57,11 +58,72 @@ const NodeListRow = memo(
     visibleIcon = VisibleIcon,
     invisibleIcon = InvisibleIcon,
     rowType,
+    focusMode,
+    parentDisabled,
+    parentPipeline,
   }) => {
-    const VisibilityIcon = checked ? visibleIcon : invisibleIcon;
+    const VisibilityIcon =
+      type === 'modularPipeline'
+        ? FocusModeIcon
+        : checked
+        ? visibleIcon
+        : invisibleIcon;
     const isButton = onClick && kind !== 'filter';
     const TextButton = isButton ? 'button' : 'div';
 
+    const determineFocusMode = (focusMode, type, id) =>
+      focusMode !== null && type === 'modularPipeline' && id === focusMode?.id;
+
+    const determineDisabledLabel = (
```

Comment:
Sorry I'd need some help understanding this bit:

Am I correct in reading this condition as a node row label is enabled if and only if either the application isn't in focused mode or its parent or itself is the currently the focused modular pipeline?

Reply:
`determineFocusMode` is the variable used to determine if the modular pipeline row _IS_ the selected modular pipeline; `determineDisabledLabel` is used to determine if the modular pipeline row belongs to part of a selected pipeline but is not the selected pipeline itself. 

and yes, your reading of this condition is correct 💯